Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -479,6 +482,25 @@ static size_t computeRangeSize(uint64_t size, size_t offset)
     indirectBufferRecomputed(0, 0, 0, 0, MTLIndexTypeUInt16, 0);
 }
 
+bool Buffer::copy(const std::span<const uint8_t> data, const size_t offset)
+{
+    auto buffer = getBufferContents();
+    if (!buffer)
+        return false;
+
+    auto endOffset = checkedSum<size_t>(offset, data.size());
+    if (endOffset.hasOverflowed() || endOffset.value() > currentSize())
+        return false;
+#if !HAVE(SWIFT_CPP_INTEROP)
+    memcpySpan(std::span<uint8_t> { buffer + offset, currentSize() - offset }, data);
+    return true;
+#else
+    return WebGPU::BufferUtils::copySpan(std::span<uint8_t> { buffer + offset, currentSize() - offset }, data);
```

Comment:
We want to structure our Swift prototype in WebGPU a little differently. Instead of unconditionally turning on the prototype when it's available, we want a separate #ifdef that controls it. This is a great patch to introduce the separate #ifdef.

The reasons we want a separate #ifdef are:
* We want to be able to A/B test prototype vs shipping for performance and correctness
* The prototype will soon use Swift features that aren't available in any shipping Swift compiler